ogr/services/github/service.py

```diff
@@ -71,13 +73,25 @@ def __init__(
         self.read_only = read_only
         self.authentication = github_authentication
 
+        if isinstance(max_retries, Retry):
+            self._max_retries = max_retries
+        else:
+            self._max_retries = Retry(
+                total=int(max_retries),
+                read=0,
+                method_whitelist=["DELETE", "GET", "PATCH", "POST", "PUT"],
+                status_forcelist=[500, 403, 401],
+                raise_on_status=False,
+            )
+
```

Comment:
I don't think this is required. `github.Github` also takes `int` or `Retry` as an argument, and `Retry` has some [sane defaults](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry). Or do you want something different then that?

Reply:
I read mentioned `Retry` (documentation)[https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry] but I wasn't satisfied with its defaults. I wanted to specify: 
- raise_on_status=False - otherwise `urllib3.exceptions.MaxRetryError` would be raised when max number of retries was reached. I wanted the retry mechanism to have minimal effects on the rest of code.
-  read=0 in order not to retry a call that could have side-effects
- status_forcelist - retry only on specific status codes

I was a quite conservative when specifying conditions when to retry - I didn't want to break anything. Other cases can be easily added in future if needed.

Reply:
I see, thank you for the explanation! Could you maybe add a comment explaining why `method_whitelist` and `status_forcelist` is different thatn the defaults. The rest of the values are kinda self-explanatory, but these two are not that ovious.

Reply:
@csomh, do you agree with changes (added comments, and 503 status code) done in 70b992e ?